connectors/aws/aws-s3/src/main/java/io/camunda/connector/aws/s3/core/S3Executor.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.aws.s3.core;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.connector.aws.CredentialsProviderSupportV2;
+import io.camunda.connector.aws.s3.model.*;
+import io.camunda.connector.aws.s3.response.DeleteResponse;
+import io.camunda.connector.aws.s3.response.DownloadResponse;
+import io.camunda.connector.aws.s3.response.UploadResponse;
+import io.camunda.document.Document;
+import io.camunda.document.store.DocumentCreationRequest;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.DeleteObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+public class S3Executor {
```

Comment:
maybe can we move the `response` package in the `model` package, not sure how we do usually